pandas/core/algorithms.py

```diff
@@ -142,6 +142,71 @@ def isin(comps, values):
     return f(comps, values)
 
 
+def safe_sort(values, labels=None, na_sentinel=-1, assume_unique=False):
+    """
+    Sort ``values`` and reorder corresponding ``labels``.
+    ``values`` should be unique if ``labels`` is not None.
+    Safe for use with mixed types (int, str), orders ints before strs.
+
+    Parameters
+    ----------
+    values : ndarray (1-d)
+        Sequence; must be unique if ``labels`` is not None.
+    labels : ndarray (1-d)
+        Indices to ``values``
+    na_sentinel : int, default -1
+        Value in ``labels`` to mark "not found".
+        Ignored when ``labels`` is None.
+    assume_unique : bool, default False
+        When True, doesn't check for uniqness of ``values``.
+        Ignored when ``labels`` is None.
+
+    Returns
+    -------
+    ordered : ndarray
+        Sorted ``values``
+    new_labels : ndarray
```

Comment:
doc string params should be like:

```
ordered: ndarray
    Sorted ``values``
new_labels: ndarray
```

e.g. no space before the colon, 1 after; certainly possible other doc-strings are wrong